src/priv.c

```
@@ -0,0 +1,258 @@
+	if (len != sizeof(*iff)) {
+		errno = EINVAL;
+		goto err;
+	}
+
+	TAILQ_FOREACH(ifp, ctx->ctx_ifaces, if_next) {
+		if (ifp->if_index == iff->if_index)
+			break;
+	}
+	if (ifp == NULL) {
+		ifp = calloc(1, sizeof(*ifp));
+		ifp->if_ctx = ctx;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)